url2df0f7c811_przemekgalazka_legacycode_tgz-pJ8-pl_cc_core_cmd_CommandJ8/src/ccproxy/ReadConfig.java

```diff
@@ -0,0 +1,226 @@
+package ccproxy;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+
+public class ReadConfig {
+	boolean configOk = false;
+	String pghost = null;
+	String pgport = "5432";
+	String pguser = null;
+	String pgpass = null;
+	String pgdbname = null;
+
+	String asteriskHost = null;
+	String asteriskPort = "5038";
+	String asteriskUser = null;
+	String asteriskSecret = null;
+	String listenPort = null;
+
+	String licensedAgents=null;
+	String licenseKey=null;
+	String licenseExpires="";
+	
+	String agentVersion = null;
+	String agentUpdateURL = null;
+	String agentContext = null;
+	// Czy po zakończeniu połączenia telefonicznego agent powienien przejść w tryb pauzy
+	boolean pauseAfterConnection = false;
+	// Czy po zalogowaniu agent powinien pozostać w trybie pauza
+	boolean pauseWhenStartup = false;
+	// Czy logować eventy płynące z asteriska.
+	boolean logAsteriskEvents = false;
+	String externAuthURL = null;
+	boolean monitorDbActivity = false;
+	boolean pauseWhenUnavailable = false;
+	boolean requestCallTag = false;
+	String configFile;
+
+	
+	String spyMode="";
+	
+	org.apache.log4j.Logger log;
+	
+
+	public ReadConfig(Parametry p, String filename) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 40. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.